dot/utils_test.go

```diff
@@ -0,0 +1,607 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package dot
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	ctoml "github.com/ChainSafe/gossamer/dot/config/toml"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/ChainSafe/gossamer/lib/utils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// newTestConfigWithFile returns a new test configuration and a temporary configuration file
+func newTestConfigWithFile(t *testing.T) (*Config, *os.File) {
+	cfg := NewTestConfig(t)
+
+	configPath := filepath.Join(cfg.Global.BasePath, "config.toml")
+	err := os.WriteFile(configPath, nil, os.ModePerm)
+	require.NoError(t, err)
+
+	cfgFile := exportConfig(cfg, configPath)
+	return cfg, cfgFile
+}
+
+// newTestGenesisFile returns a human-readable test genesis file using "gssmr" human readable data
+func newTestGenesisFile(t *testing.T, cfg *Config) (filename string) {
+	fp := utils.GetGssmrGenesisPathTest(t)
+
+	gssmrGen, err := genesis.NewGenesisFromJSON(fp, 0)
+	require.NoError(t, err)
+
+	gen := &genesis.Genesis{
+		Name:       cfg.Global.Name,
+		ID:         cfg.Global.ID,
+		Bootnodes:  cfg.Network.Bootnodes,
+		ProtocolID: cfg.Network.ProtocolID,
+		Genesis:    gssmrGen.GenesisFields(),
+	}
+
+	b, err := json.Marshal(gen)
+	require.NoError(t, err)
+
+	filename = filepath.Join(t.TempDir(), "genesis.json")
+	err = os.WriteFile(filename, b, os.ModePerm)
+	require.NoError(t, err)
+
+	return filename
+}
+
+func TestCreateJSONRawFile(t *testing.T) {
+	type args struct {
+		bs *BuildSpec
+		fp string
+	}
+	tests := []struct {
+		name         string
+		args         args
+		expectedHash string
+	}{
+		{
+			name: "working example",
+			args: args{
+				bs: &BuildSpec{genesis: NewTestGenesis(t)},
+				fp: filepath.Join(t.TempDir(), "/test.json"),
+			},
+			expectedHash: "23356cdb5d3537d39b735726707216c9e329c7b8a2c8a41b25da0f5f936b3caa",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := CreateJSONRawFile(tt.args.bs, tt.args.fp)
+			file, err := os.Open(got.Name())
+			require.NoError(t, err)
+			defer file.Close()
+
+			buf := make([]byte, 30*1024)
+			sha256 := sha256.New()
+			for {
+				n, err := file.Read(buf)
+				if n > 0 {
+					_, err := sha256.Write(buf[:n])
+					require.NoError(t, err)
+				}
+				if err == io.EOF {
+					break
+				}
+				if err != nil {
+					log.Errorf("read %d bytes: %v", n, err)
+					break
+				}
+			}
+			sum := sha256.Sum(nil)
+			require.Equal(t, tt.expectedHash, fmt.Sprintf("%x", sum))
```

Comment:
some fancy performance code you have here, but that just works for a test :wink: 

```suggestion

			b, err := ioutil.ReadFile(got.Name())
			require.NoError(t, err)

			digest := sha256.Sum256(b)
			hexDigest := fmt.Sprintf("%x", digest)

			require.Equal(t, tt.expectedHash, hexDigest)
```